// @storybook/addon-info/dist/components/markdown/code.js
"use strict";

require("core-js/modules/es.object.define-property");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Blockquote = Blockquote;
Object.defineProperty(exports, "Pre", {
  enumerable: true,
  get: function get() {
    return _pre["default"];
  }
});
exports.Code = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _components = require("@storybook/components");

var _pre = _interopRequireDefault(require("./pre/pre"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var Code = function Code(_ref) {
  var language = _ref.language,
      code = _ref.code;
  return _react["default"].createElement(_components.SyntaxHighlighter, {
    bordered: true,
    copyable: true,
    language: language
  }, code);
};

exports.Code = Code;
Code.propTypes = {
  language: _propTypes["default"].string.isRequired,
  code: _propTypes["default"].string.isRequired
};

function Blockquote(_ref2) {
  var children = _ref2.children;
  var style = {
    fontSize: '1.88em',
    fontFamily: 'Menlo, Monaco, "Courier New", monospace',
    borderLeft: '8px solid #fafafa',
    padding: '1rem'
  };
  return _react["default"].createElement("blockquote", {
    style: style
  }, children);
}

Blockquote.propTypes = {
  children: _propTypes["default"].node
};
Blockquote.defaultProps = {
  children: null
};
